hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/BaseRollbackActionExecutor.java

```diff
@@ -261,7 +261,10 @@ protected void finishRollback(HoodieInstant inflightInstant, HoodieRollbackMetad
       // If publish the rollback to the timeline, we finally transition the inflight rollback
       // to complete in the data table timeline
       if (!skipTimelinePublish) {
-        table.getActiveTimeline().transitionRollbackInflightToComplete(inflightInstant,
+        // NOTE: no need to lock here, since !skipTimelinePublish is always true,
+        // when skipLocking is false, txnManager above-mentioned should lock it.
+        // when skipLocking is true, the caller should have already held the lock.
```

Comment:
esp stuff like this, it's better to have re-entrancy here as opposed to avoiding locks, so we don't introduce code deps on locking, which is hard to debug.

Reply:
re-entrancy is good but you still need to keep track of the holdness of the lock explicitly, we have a `LockManager` in the `TxnManager`, but the `LockManager` is only available in the write client, and it is not designed as a singleton now.